src/boot/bootctl.c

```diff
@@ -112,6 +113,20 @@ static int acquire_xbootldr(bool unprivileged_mode, sd_id128_t *ret_uuid) {
         return 1;
 }
 
+static void settle_make_machine_id_directory(void) {
+        struct statfs stf;
+
+        if (arg_make_machine_id_directory != -1)
+                return;
+
+        arg_make_machine_id_directory = false;
+
+        if (statfs("/etc/machine-id", &stf) < 0)
+                return;
+
+        arg_make_machine_id_directory = !is_temporary_fs(&stf);
```

Comment:
```c
r = path_is_temporary_fs("/etc/machine-id");
if (r < 0)
        log_debug_errno(r, "Couldn't determine whether /etc/machine-id is on a temporary file system, assuming it is not.");

arg_make_machine_id_directory = r <= 0;
```

(i.e. use path_is_temporary_fs() instead of is_temproary_fs(), and please add debug logging for unexpected errors, even if we then ignore them)

Reply:
Right. I went with different logic since, well, `statfs("/etc/machine-id")` will reasonably only fail if `/etc/machine-id` doesn't exist, which is the highest state of ephemerality.